sydent/replication/peer.py

```diff
@@ -83,16 +82,25 @@ def verifyMessage(self, jsonMessage):
         if not 'signatures' in jsonMessage:
             raise NoSignaturesException()
 
+        alg = 'ed25519'
+
         key_ids = signedjson.sign.signature_ids(jsonMessage, self.servername)
-        if not key_ids or len(key_ids) == 0 or not key_ids[0].startswith("ed25519:"):
+        if not key_ids or len(key_ids) == 0 or not key_ids[0].startswith(alg + ":"):
             e = NoMatchingSignatureException()
             e.foundSigs = jsonMessage['signatures'].keys()
             e.requiredServername = self.servername
             raise e
-        verify_key = yield self.get_server_verify_key(server_name, key_ids)
-        verifyKey = nacl.signing.VerifyKey(self.pubkeys['ed25519'], encoder=nacl.encoding.HexEncoder)
-        verifyKey.alg = 'ed25519'
-        signedjson.sign.verify_signed_json(jsonMessage, self.servername, verifyKey)
+
+        # Get verify key from signing key
+        signing_key = signedjson.key.decode_signing_key_base64(alg, "0", self.pubkeys[alg])
```

Comment:
could we do this (the calculation of verify_key) in the constructor rather than on every message?